### ISSUE TITLE
Multiple unstake postpones refund indefinitely

### ISSUE THREAD

**Issue by jhsikkema** [EOSIO/eos#8185]
I performed several unstake transactions. The behavior one would expect would be that the eos that was unstaked more than 72 hours ago would be refunded, however it the unstaked eos is not released until the last transaction passed the 72 hour limit. This allows for an indefinite time for the refund to take place, which is a violation of the promised 3 day period, also this is for me very counter intuitive and makes it much harder to program a dynamic staking/unstaking algorithm. 

The associated transactions:
1.
https://bloks.io/transaction/b8c9fd71473e85cac125fda7460c659903f1fa6d90ac69abd5e3aa1417d0cd91
Timestamp: Nov-01-2019, 07:42:21 PM UTC

2.
https://bloks.io/transaction/98ea397682273d8cb15daebc25896e9d5451f1c4c052d8337c482fe42fb1097
Timestamp: Nov-01-2019, 08:15:36 PM UTC

3. 
https://bloks.io/transaction/a376af6792ace973b2a8c750f2006ff22227239bd9e677fed0a420454849e7aa
at Nov-03-2019, 03:38:47 PM UTC

Output of refund:
Error 3050003: eosio_assert_message assertion failure
Error Details:
assertion failure with message: refund is not available yet
pending console output:

**Comment by taokayan** [comment]
Did you trigger another unstake during the 3-day waiting period?

**Comment by SonicAgamemnon** [comment]
I bet he did just that.  I've tried it myself and can confirm that if you unstake various amounts several times following an initial unstake, no funds are released until 3 days after the last unstake event took place.

**Comment by jhsikkema** [comment]
Yes, as I describe in the issue above I triggered multiple unstakes in the 3-day waiting period. Now for me this behavior of resetting the waiting period of previous unstakes is simply unacceptable. It is advertised that the staked funds will be released in three days, but on average I experience it takes more like 4 or 5 days (either I need to wait for a current unstake to be processed before I can unstake again or I need to reset the unstakes already in process). This makes the advertised waiting period of three days misleading and in my view this is precisely the kind of thing here in Europe might get into trouble with regulation for small investors (roughly more than 100 investors with less than EUR 100.000 invested). Now one way to go would be to update the documentation, but I think it will make eosio actually better if the unstaking mechanism is fixed.

**Comment by feri42** [comment]
I find this behaviour unacceptable too. It effectively holds my EOS hostage and makes my account useless for 3 days.

**Comment by damsalem** [comment]
This message will hopefully help others who can't get their unstaked tokens. EOSAuthority.com has a crap refund process. I successfully triggered a manual refund through Bloks.io. It was instantaneous. You'll need a wallet already set up (Scatter, Ledger, etc.)
